test/unit/jit/solver/test_jit_solver_builder.cpp

```diff
@@ -0,0 +1,69 @@
+#include <micm/solver/backward_euler.hpp>
+#include <micm/solver/rosenbrock.hpp>
+#include <micm/solver/rosenbrock_solver_parameters.hpp>
+
+#include <micm/jit/solver/jit_solver_builder.hpp>
+#include <micm/jit/solver/jit_solver_parameters.hpp>
+
+#include <micm/solver/solver_builder.hpp>
+#include <micm/util/matrix.hpp>
+#include <micm/util/sparse_matrix.hpp>
+#include <micm/util/sparse_matrix_vector_ordering.hpp>
+#include <micm/util/vector_matrix.hpp>
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <random>
+
+namespace
+{
+  auto a = micm::Species("A");
+  auto b = micm::Species("B");
+  auto c = micm::Species("C");
+
+  micm::Phase gas_phase{ std::vector<micm::Species>{ a, b, c } };
+
+  micm::Process r1 = micm::Process::Create()
+                         .SetReactants({ a })
+                         .SetProducts({ micm::Yields(b, 1) })
+                         .SetRateConstant(micm::ArrheniusRateConstant({ .A_ = 2.15e-11, .B_ = 0, .C_ = 110 }))
+                         .SetPhase(gas_phase);
+
+  micm::Process r2 = micm::Process::Create()
+                         .SetReactants({ b })
+                         .SetProducts({ micm::Yields(c, 1) })
+                         .SetRateConstant(micm::ArrheniusRateConstant(
+                             micm::ArrheniusRateConstantParameters{ .A_ = 3.3e-11, .B_ = 0, .C_ = 55 }))
+                         .SetPhase(gas_phase);
+  micm::System the_system = micm::System(micm::SystemParameters{ .gas_phase_ = gas_phase });
+  std::vector<micm::Process> reactions = { r1, r2 };
+}  // namespace
+
+TEST(SolverBuilder, CanBuildJitRosenbrock)
+{
+  constexpr std::size_t L = 4;
+  auto jit_rosenbrock = micm::JitSolverBuilder<micm::JitRosenbrockSolverParameters, L>(
+                            micm::RosenbrockSolverParameters::ThreeStageRosenbrockParameters())
+                            .SetSystem(the_system)
+                            .SetReactions(reactions)
+                            .SetNumberOfGridCells(L)
+                            .Build();
+}
+
+TEST(SolverBuilder, CanBuildCudaSolvers)
+{
+  // auto cuda_backward_euler = micm::CudaSolverBuilder<L>()
+  //                           .SetSystem(the_system)
+  //                           .SetReactions(reactions)
+  //                           .SetNumberOfGridCells(1)
+  //                           .SolverParameters(micm::BackwardEulerSolverParameters{})
+  //                           .Build();
+
+  // auto cuda_rosenbrock = micm::CudaSolverBuilder<L>()
+  //                           .SetSystem(the_system)
+  //                           .SetReactions(reactions)
+  //                           .SetNumberOfGridCells(1)
+  //                           .SolverParameters(micm::ThreeStageRosenbockSolverParameters{})
+  //                           .Build();
```

Comment:
can this be uncommented now? or is there still more to do for building CUDA Rosenbrock solvers?

Reply:
This test does exist in test/unit/cuda/test_cuda_solver_builder.cpp. This file is in the jit and we don't need this code